overlays/tools/get-changed-modules-push.sh

```diff
@@ -0,0 +1,199 @@
+#-----------------------------------------------------------------------------------------                   
+function usage {
+    info "Syntax: get-changed-modules-push.sh [OPTIONS]"
+    cat << EOF
+Options are:
+-h | --help : Display this help text
+--repo The repository for this GitHub Actions workflow
+--base The base commit level (the commit before the push)
+--head The head commit level (the commit that triggered the push)
+EOF
+}
+
+module_names=(\
+    "buildutils" \
+    "wrapping" \
+    "gradle" \
+    "maven" \
+    "framework" \
+    "extensions" \
+    "managers" \
+    "obr" \
+)
+
+#-----------------------------------------------------------------------------------------
+# Process parameters
+#-----------------------------------------------------------------------------------------
+repo=""
+base=""
+head=""
+while [ "$1" != "" ]; do
+    case $1 in
+        -h | --help )   usage
+                        exit
+                        ;;
+
+        --repo )        repo="$2"
+                        shift
+                        ;;
+
+        --base )        base="$2"
+                        shift
+                        ;;
+
+        --head )        head="$2"
+                        shift
+                        ;;
+
+        * )             error "Unexpected argument $1"
+                        usage
+                        exit 1
+    esac
+    shift
+done
+
+#-----------------------------------------------------------------------------------------
+# Functions
+#-----------------------------------------------------------------------------------------
+
+function get_files_changed_in_push() {
+
+    h1 "Getting the files changed in Push event triggered by commit ${head}" 
+
+    h2 "Files changed:"
+    gh api repos/${repo}/compare/${base}...${head} --jq '.files[].filename'
+
+    # Extract changed module names from changed files from GitHub CLI output
+    mapfile -t changed_files_in_push < <(gh api repos/${repo}/compare/${base}...${head} --jq '.files[].filename')
+
+    modules_changed_in_push=()
+    for changed_file in "${changed_files_in_push[@]}"; do
+        echo "$changed_file"
+        module=$(echo "$changed_file" | cut -d'/' -f2)
+        modules_changed_in_push+=("$module")
+    done
+
+    # Remove possible duplicates from array of changed modules
+    declare -A unique_module_map
+
+    unique_modules_found_in_push=()
+    for module in "${modules_changed_in_push[@]}"; do
+    if [[ -z "${unique_module_map[$module]}" ]]; then
+        unique_modules_found_in_push+=("$module")
+        unique_module_map[$module]=1
+    fi
+    done
+
+    h2 "Modules changed:"
+    echo "${unique_modules_found_in_push[@]}"
+
+}
+
+function get_changed_modules_and_set_in_environment() {
+
+    h1 "Finding changed modules and setting environment variables that can be used in the GitHub Actions workflows..."
+
+    for module in "${unique_modules_found_in_push[@]}"; do
+        if [[ "$module" == "buildutils" ]]; then
```

Comment:
same comment as above.

If this is a duplicate function, I would put it into a second script and call that ?